repo.go

```diff
@@ -671,6 +672,38 @@ func (r *Repo) AddTargets(paths []string, custom json.RawMessage) error {
 	return r.AddTargetsWithExpires(paths, custom, data.DefaultExpires("targets"))
 }
 
+func (r *Repo) AddDigestTargets(digest string, length int64, custom json.RawMessage) error {
+	expires := data.DefaultExpires("targets")
+
+	t, err := r.targets()
+	if err != nil {
+		return err
+	}
+	hashes := make([]string, 1)
+	split_digest := strings.Split(digest, ":")
+	hashes[0] = split_digest[1]
```

Comment:
We probably should return an error here if `digest` is not the proper format, otherwise this could panic.